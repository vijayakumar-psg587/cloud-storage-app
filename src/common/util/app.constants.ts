export const APP_NAME = 'CLOUD_AUTHENTICATOR';
export const APP_DEFAULT_TIMESTAMP_FORMAT = 'YYYY-DD-MM hh:mm:ss.SSSS';
export const APP_SERVER_ZONE  = 'America/New_York';
export const CHAR_HYPEN = '-';
export const CHAR_ZERO = '0';
export const CHAR_COLON = ':';
export const APP_MESSAGE_KEY = 'cloud-storage-message-key';
export const APP_LOG_REDACT = [];
export const APP_LOG_CENSOR = '**REDACTED';
export const GCLOUD_KEY_FILE = '/home/vijaykumar/Documents/glcoud/service-accounts/dev-index-254608-03_17_2020.json';
export const GCLOUD_PRJ_ID = 'dev-index-254608';
export const GCLOUD_STROAGE_SCOPE = ['https://www.googleapis.com/auth/devstorage.read_write','https://www.googleapis.com/auth/pubsub'];
export const GCLOUD_CONN_MAX_RETRIES = 3;
export const GCLOUD_ENV_STR_BUCKET_NAME = 'feedback-database-connector';
export const GCLOUD_UPLOAD_FILE_DEV_LOCAL_PATH = '/home/vijaykumar/Documents/Code/Nestjs/database-connector/config/development/dev.txt';
export const GCLOUD_UPLOAD_FILE_PROD_LOCAL_PATH = '/home/vijaykumar/Documents/Code/Nestjs/database-connector/config/production/prod.txt';
export const GCLOUD_DATABASE_BUCKET_DEV = 'mongo-data/development/dev.txt';
export const GCLOUD_DATABASE_BUCKET_PROD = 'mongo-data/production/prod.txt';
export const ENV_NAME_DEV = 'dev';
export const GCLOUD_STORED_FILE_NAME_DEV = 'dev.txt';
export const GCLOUD_STORED_FILE_NAME_PROD = 'prod.txt';

