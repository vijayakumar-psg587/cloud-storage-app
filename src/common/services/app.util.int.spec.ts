import { AppUtilServiceInstance } from './app.util.service';
import moment from 'moment';
import mock from 'mock-fs';
import { AuthenticationModel } from '../models/authentication.model';

describe('New Test suite for util service', () => {
	beforeEach(() => {
		jest.restoreAllMocks();
	});
	afterAll(() => {
		jest.clearAllMocks();
		jest.resetModules();
		mock.restore();
	});

	test('testing general getUTC time -  actual momentimezone call', () => {
		const timestamp = AppUtilServiceInstance.getUTCTimeStamp('Asia/Calcutta');
		const expectedDateString = moment(new Date(Date.now()).toISOString())
			.tz('Asia/Calcutta')
			.format('YYYY-DD-MM hh:mm:ss.SSSS');
		expect(timestamp).toBeDefined();
		expect(timestamp).toMatchSnapshot(expectedDateString);
	});

	test('testing getGcloudAuthModel method', () => {
		const resultModel: AuthenticationModel = AppUtilServiceInstance.getGcloudAuthModel();
		expect(resultModel).toBeDefined();
	});

	test('testing getFile method with input', () => {
		const result = AppUtilServiceInstance.getFileName('/test/path/fileName.ts');
		const expectedResult = 'fileName';
		expect(result).toEqual(expectedResult);
	});

  test('testing getFile method w/o input', () => {
    const result = AppUtilServiceInstance.getFileName(null);
    const expectedResult = 'Unnamed file';
    expect(result).toEqual(expectedResult);
  });
});
