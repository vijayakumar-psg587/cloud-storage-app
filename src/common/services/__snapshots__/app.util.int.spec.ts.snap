// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Test suite for util service testing general getUTC time -  actual momentimezone call: 2020-21-04 02:08:25.7410 1`] = `"2020-21-04 02:08:25.7390"`;

exports[`New Test suite for util service testing general getUTC time -  actual momentimezone call: 2020-21-04 02:08:57.3920 1`] = `"2020-21-04 02:08:57.3910"`;

exports[`New Test suite for util service testing general getUTC time -  actual momentimezone call: 2020-21-04 02:13:50.6630 1`] = `"2020-21-04 02:13:50.6630"`;
